File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/ComponentDescriptor.java
package com.enonic.wem.api.content.page;


import com.enonic.wem.api.form.Form;
import com.enonic.wem.api.module.ModuleResourceKey;

public interface ComponentDescriptor
{
    ComponentDescriptorName getName();

    String getDisplayName();

    Form getConfig();

    ModuleResourceKey getControllerResource();
}

File: modules/wem-webapp/src/main/webapp/admin/common/js/content/page/ComponentDescriptor.ts
module api.content.page{

    export class ComponentDescriptor{

        private name:ComponentDescriptorName;

        private displayName:string;

        private form:api.form.Form;

        constructor(builder:ComponentDescriptorBuilder) {
            this.name = builder.name;
            this.displayName = builder.displayName;
            this.form = builder.form;
        }

        getName():ComponentDescriptorName {
            return this.name;
        }

        getDisplayName():string {
            return this.displayName;
        }

        getForm():api.form.Form {
            return this.form;
        }
    }

    export class ComponentDescriptorBuilder {

        name:ComponentDescriptorName;

        displayName:string;

        form:api.form.Form;
    }
}